prow/jira/jira.go

```diff
@@ -259,8 +288,275 @@ func (jc *client) UpdateRemoteLink(id string, link *jira.RemoteLink) error {
 	return nil
 }
 
-func (jc *client) JiraURL() string {
-	return jc.url
+func (jc *client) DeleteRemoteLink(issueID string, linkID int) error {
+	apiEndpoint := fmt.Sprintf("/rest/api/2/issue/%s/remotelink/%d", issueID, linkID)
+	req, err := jc.upstream.NewRequest("DELETE", apiEndpoint, nil)
+	if err != nil {
+		return err
+	}
+
+	// the response should be empty if it is not an error
+	emptyStruct := struct{}{}
+	resp, err := jc.upstream.Do(req, &emptyStruct)
+	// status code 204 is a success for this function...
+	if resp.StatusCode != 204 && err != nil {
+		return JiraError(resp, err)
+	}
+	return nil
+}
+
+// GenericDeleteRemoteLinkViaURL identifies and removes a remote link from an issue
+// the has the provided URL.
+func GenericDeleteRemoteLinkViaURL(jc Client, issueID, url string) error {
```

Comment:
Why do these have `Generic` in their name? You can just omit it, it won't clash with the method names